### PR TITLE
{vis}[foss/2018b,intel/2018b] IPython v5.8.0

### PR DESCRIPTION
**Pull request by jose-d**
IPython for Python 2.7.15 intel 2018b..

(created using `eb --new-pr`)

### CODE REVIEW

**Comment by jose-d** [comment]
Test report by @jose-d
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
koios2 - Linux centos linux 7.3.1611, Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz, Python 2.7.5
See https://gist.github.com/5c82a5560c0594b21ec4947ce167bd92 for a full test report.

**Comment by jose-d** [comment]
```
found 3 variants of 'matplotlib' dependency in easyconfigs using '2018b' toolchain generation
* version: 2.2.3; versionsuffix: -Python-2.7.15 as dep for set(['Seaborn-0.9.0-foss-2018b-Python-2.7.15.eb', 'molmod-1.4.4-intel-2018b-Python-2.7.15.eb', 'OptiType-1.3.2-foss-2018b-Python-2.7.15.eb', 'Amber-18-fosscuda-2018b-AmberTools-18-patchlevel-10-8.eb', 'VCF-kit-0.1.6-intel-2018b-Python-2.7.15.eb', 'yaff-1.4.5-intel-2018b-Python-2.7.15.eb', 'PyNAST-1.2.2-foss-2018b-Python-2.7.15.eb', 'Amber-18-foss-2018b-AmberTools-18-patchlevel-10-8.eb', 'Assimulo-2.9-intel-2018b-Python-2.7.15.eb', 'BlobTools-20180528-foss-2018b-Python-2.7.15.eb', 'HTSeq-0.11.0-foss-2018b-Python-2.7.15.eb', 'PyCogent-1.9-foss-2018b-Python-2.7.15.eb', 'HTSeq-0.11.0-intel-2018b-Python-2.7.15.eb'])
* version: 2.2.4; versionsuffix: -Python-2.7.15 as dep for set(['IPython-5.8.0-intel-2018b-Python-2.7.15.eb'])
* version: 3.0.0; versionsuffix: -Python-3.6.6 as dep for set(['ASAP3-3.10.10-foss-2018b-Python-3.6.6.eb', 'QuTiP-4.3.1-foss-2018b-Python-3.6.6.eb', 'ASE-3.17.0-foss-2018b-Python-3.6.6.eb', 'HDDM-0.6.1-intel-2018b-Python-3.6.6.eb', 'IPython-7.2.0-foss-2018b-Python-3.6.6.eb', 'ASE-3.16.2-intel-2018b-Python-3.6.6.eb', 'basemap-1.2.0-intel-2018b-Python-3.6.6.eb', 'IPython-7.2.0-fosscuda-2018b-Python-3.6.6.eb', 'ASE-3.16.2-foss-2018b-Python-3.6.6.eb', 'scikit-learn-0.20.2-foss-2018b-Python-3.6.6.eb', 'ASAP3-3.10.10-iomkl-2018b-Python-3.6.6.eb', 'scikit-learn-0.20.0-foss-2018b-Python-3.6.6.eb', 'scikit-image-0.14.1-intel-2018b-Python-3.6.6.eb', 'HTSeq-0.11.2-foss-2018b-Python-3.6.6.eb', 'GPAW-1.4.0-intel-2018b-Python-3.6.6.eb', 'MultiQC-1.7-foss-2018b-Python-3.6.6.eb', 'ASE-3.17.0-iomkl-2018b-Python-3.6.6.eb', 'FMRIprep-1.1.8-foss-2018b-Python-3.6.6.eb', 'ASAP3-3.10.10-intel-2018b-Python-3.6.6.eb', 'ICA-AROMA-0.4.4-beta-foss-2018b-Python-3.6.6.eb', 'Seaborn-0.9.0-foss-2018b-Python-3.6.6.eb', 'Spyder-3.3.2-foss-2018b-Python-3.6.6.eb', 'ASE-3.17.0-intel-2018b-Python-3.6.6.eb', 'scikit-image-0.14.1-foss-2018b-Python-3.6.6.eb', 'scikit-optimize-0.5.2-intel-2018b-Python-3.6.6.eb', 'GPAW-1.4.0-foss-2018b-Python-3.6.6.eb', 'QuTiP-4.3.1-intel-2018b-Python-3.6.6.eb', 'ASE-3.16.2-iomkl-2018b-Python-3.6.6.eb', 'IPython-7.2.0-intel-2018b-Python-3.6.6.eb', 'imbalanced-learn-0.4.3-foss-2018b-Python-3.6.6.eb', 'scikit-learn-0.20.0-fosscuda-2018b-Python-3.6.6.eb', 'iCount-20180820-foss-2018b-Python-3.6.6.eb', 'scikit-image-0.14.1-fosscuda-2018b-Python-3.6.6.eb', 'scikit-learn-0.20.0-intel-2018b-Python-3.6.6.eb'])
```
does it mean there is limited amount of possible variants of matplotlib per (this particular) toolchain?

**Comment by akesandgren** [comment]
We try to stick to a single version per toolchain for things that are used as dependencies of higher level software. The reason is to minimize the risk of version conflicts when loading multiple user-level softwares.

**Comment by akesandgren** [comment]
@jose-d We need to synchronize this PR's ext_list with https://github.com/easybuilders/easybuild-easyconfigs/pull/7647

**Comment by jose-d** [comment]
> @jose-d We need to synchronize this PR's ext_list with #7647
> 
> Do you have time to go through the lists and let me know what the diffs are and verify the the versions used are ok, i.e. not too old or too new in any given case?

yup, I think in second half of this week I could take a look.

**Comment by jose-d** [comment]
> Do you have time to go through the lists and let me know what the diffs are and verify the the versions used are ok, i.e. not too old or too new in any given case?

@akesandgren : mhmm, done: https://github.com/easybuilders/easybuild-easyconfigs/pull/7805/commits/8dce0f7310d6a42a605a1b1cc0f6823c4c2eebd6

**Comment by akesandgren** [comment]
Ehhh?? Did you just take what I had in 7647 straight off?
I'm not sure if that's a good list with good versions for IPython 5.8.0.

I was hoping that you'd do the work of making sure that
1 - both PR's include the same list of extensions
2 - that the version of the extensions used are the same and good versions are used...

I just picked whatever was there at the time that didn't have a hard requirement and based it on IPythyon 5.7.0's list.

But I'm not picky... 

Are they the same now? Haven't checked (way overloaded with work)...

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 5 out of 5 (2 easyconfigs in this PR)
b-an03.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) CPU E5-2690 v4 @ 2.60GHz, Python 2.7.12
See https://gist.github.com/5d3b399f38354eb8f661ea002b68541b for a full test report.

**Comment by akesandgren** [comment]
We'll hold off on merging this untill @migueldiascosta has run a testreport on https://github.com/easybuilders/easybuild-easyconfigs/pull/7647

**Comment by jose-d** [comment]
> We'll hold off on merging this untill @migueldiascosta has run a testreport on #7647

yep. as this PR already contains 5 commits, perhaps it'd be better to open new one to do not mess the upstream git tree.. ok?

**Comment by akesandgren** [comment]
No, don't change anything now. Let's wait for miguel's test report and then we will move forward with these two as they are.

**Comment by jose-d** [comment]
@akesandgren I saw @migueldiascosta already merged #7647. 
Is anything here remaining to be done from my side?

File: easybuild/easyconfigs/i/IPython/IPython-5.8.0-foss-2018b-Python-2.7.15.eb [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+    }),
+    ('backports.shutil_get_terminal_size', '1.0.0', {
+        'source_urls': ['https://pypi.python.org/packages/source/b/backports.shutil_get_terminal_size/'],
+        'checksums': ['713e7a8228ae80341c70586d1cc0a8caa5207346927e23d09dcbcaf18eadec80'],
+    }),
+    ('testpath', '0.4.2', {
+        'source_urls': ['https://pypi.python.org/packages/source/t/testpath/'],
+        'checksums': ['b694b3d9288dbd81685c5d2e7140b81365d46c29f5db4bc659de5aa6b98780f8'],
+    }),
+    ('nose', '1.3.7', {
```
Comment (boegel) [inline easybuild/easyconfigs/i/IPython/IPython-5.8.0-foss-2018b-Python-2.7.15.eb]: @jose-d These easyconfigs include a lot of additional extensions compared to #7647, even things already included with `Python` itself, can you clarify why that is?

Comment (jose-d) [reply easybuild/easyconfigs/i/IPython/IPython-5.8.0-foss-2018b-Python-2.7.15.eb]: it's long time ago already..., I'll check it.

**Comment by jose-d** [comment]
Test report by @jose-d
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
koios1 - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6130 CPU @ 2.10GHz, Python 2.7.5
See https://gist.github.com/529c0ebbf4ebda55ac23fb04e3e5d6b9 for a full test report.

File: easybuild/easyconfigs/i/IPython/IPython-5.8.0-foss-2018b-Python-2.7.15.eb (hunk 1 above)
Comment (jose-d) [reply easybuild/easyconfigs/i/IPython/IPython-5.8.0-foss-2018b-Python-2.7.15.eb]: synced from 7647.
> These easyconfigs include a lot of additional extensions
there was just different order as both PRs were originally independent.
> even things
mhmm, I found just `certifi` and removed it from this PR.

**Comment by jose-d** [comment]
@boegel , @migueldiascosta :
is here anything to be done from my side?

**Comment by boegel** [comment]
Sorry for the holdup @jose-d, it looks good to go, will merge as soon as I've submitted a test report (or a couple even).

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node3105.skitty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 2.7.5
See https://gist.github.com/e2643210c82f8120f811bd571c3b940c for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2661.swalot.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/11a292b2dceb74b9dd5a0dfbd25573c7 for a full test report.

**Comment by boegel** [comment]
(close/re-open to re-trigger more extensive tests in Travis after the `4.x` branch was collapsed into `develop`)

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2113.delcatty.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/749af79b9d8ab0432d936e1408212768 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @jose-d!

**Pull request merged**